### PR TITLE
add portmap for docker agent

### PR DESCRIPTION
**Pull request by dexhunter**
* related to #254
* add ports options when creating docker agent

Signed-off-by: Dixing Xu <dixingxu@gmail.com>

### CODE REVIEW

File: src/agent/docker-rest-agent/server.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def create_node():
 
     try:
         # same as `docker run -dit yeasy/hyperledge-fabric:2.2.0 -e VARIABLES``
-        container = client.containers.run(request.form.get('img'), request.form.get('cmd'), detach=True, tty=True, stdin_open=True, name=request.form.get('name'), environment=env)
+        container = client.containers.run(request.form.get('img'), request.form.get('cmd'), detach=True, tty=True, stdin_open=True, name=request.form.get('name'), environment=env, ports=request.form.get('port_map'))
```
Comment (yeasy) [inline src/agent/docker-rest-agent/server.py:46]: And the port mapping should be sent to the docker daemon to take effect?

Comment (dexhunter) [reply src/agent/docker-rest-agent/server.py:46]: I am not sure if I understand the question correctly. From what I read on [the doc](https://docs.google.com/document/d/1XwkUORuySV2uDo1tl2e3SA3RPwrsKCoVplqmmKiK9N0/edit#heading=h.6e6uni3nrh5a), we pass `port-map` as a mapping dict from public to internal services, and this `ports(*dict*)` parameter binds the ports insides the container as listed in [this doc](https://docker-py.readthedocs.io/en/stable/containers.html#module-docker.models.containers) . For example, we can pass `{'7053/tcp': 8053}` which will expose port 7053 inside the container as port 8053 on the host.

Or am I understanding this feature wrong? Do you want to use `port-map` to configure the corresponding blockchain services? Maybe we can change that on the `config-file`?

Comment (XuHugo) [reply src/agent/docker-rest-agent/server.py:46]: I think what he understands is correct. The original port of the node (defined in config. Yaml) does not change. Let's map it to the port of the host;

Comment (yeasy) [reply src/agent/docker-rest-agent/server.py:46]: In this case, we still need to specify the host port via the docker client? Or this patchset is enough?

Comment (XuHugo) [reply src/agent/docker-rest-agent/server.py:46]: i think this patchset is enough.

**Review by yeasy — APPROVED** [review]

**Pull request merged**